R/FilterStates.R

```diff
@@ -161,25 +140,16 @@ FilterStates <- R6::R6Class( # nolint
         # below code translates to call by the names of filter_items
         rhs <- call_with_colon(
           self$get_fun(),
-          private$input_dataname,
+          `if`(is.call(private$dataname), private$dataname, as.name(private$dataname)),
```

Comment:
if dataname is only a character then we don't need this? And on line 148

Reply:
Reason is MAE - in this case it still can be a call. BUT - maybe it would be better to reverse it and create a custom get_call for FilterStatesMAE.R.

Reply:
at the very least add a comment here so when I look next time I see why it's like this :)

Reply:
Why is custom call needed? If we have `dataname = character(1)` in all filter-states then we can still use abstract `get_call`, right? If you need to turn `character` to call please just use `str2lang()`

`language` is used only in `get_call` methods and we needed to convert dataname into character using `deparse1(private$input_dataname)` everywhere else. Point of the issue is to simplify the classes - let's keep this field as a `character` only and use `character` only to initialize the class.

Just a brief look how faster is turning from string to language. Comparison is still on the level of microseconds, but still shows that keeping `private$dataname = character(1)` and converting it to the `language` when using `get_call` is not a big deal.

```r
call <- quote(MAE[[xxx]])
microbenchmark::microbenchmark(
  str2lang("MAE[[xxx]]"),
  deparse1(call)
)
#                   expr    min     lq     mean median     uq    max neval
# str2lang("MAE[[xxx]]")  1.401  1.601  1.83901  1.701  1.801 12.201   100
#         deparse1(call) 15.500 15.901 17.13894 16.201 16.751 74.501   100
```